bazel/envoy_build_system.bzl

```diff
@@ -213,6 +214,11 @@ def envoy_cmake_external(
     # On Windows, we don't want to explicitly set CMAKE_BUILD_TYPE,
     # rules_foreign_cc will figure it out for us
     cache_entries_no_build_type = {key: cache_entries[key] for key in cache_entries.keys() if key != "CMAKE_BUILD_TYPE"}
+    cache_entries_no_build_type_debug = dict(cache_entries_no_build_type)
+    cache_entries_debug = dict(cache_entries)
+    for key in debug_cache_entries.keys():
```

Comment:
Nit: you should just be able to do `foo.update(bar)` for dictionary update in Python/Starlark.